process_request/helpers.py

```diff
@@ -3,6 +3,164 @@
 from rapidfuzz import fuzz
 
 
+def get_container_indicators(instance):
+    """Takes ArchivesSpace JSON instance data with resolved top containers and returns a container indicator.
+
+    Args:
+        instance (dict): ArchivesSpace instance information that has resolved top containers and digital objects.
+
+    Returns:
+        string: A concatenated string containing the container type and container indicator, or digital object title.
+    """
+    if instance.get("instance_type") == "digital_object":
+        return "Digital Object: " + instance.get("digital_object").get("_resolved").get("title")
+    else:
+        return instance.get("sub_container").get("top_container").get("_resolved").get("type").capitalize() + ' ' + instance.get("sub_container").get("top_container").get("_resolved").get("indicator")
+
+
+def get_file_versions(digital_object):
+    """Takes ArchivesSpace digital object json and returns a file version.
+
+    Args:
+        digital_object (dict): Resolved json of an ArchivesSpace digital object.
+
+    Returns:
+        string: all file version uris associated with the digital object, separated by a comma.
+    """
+    versions = []
+    [versions.append(v.get("file_uri")) for v in digital_object.get("file_versions")]
+    return ", ".join(versions)
+
+
+def get_instance_data(instance, type):
+    """Takes ArchivesSpace instance information and returns a dictionary of selected instance information.
+
+    Args:
+        instance (dict): ArchivesSpace instance information that has resolved top containers and digital objects.
+        type (string): a string representation of an instance type.
+
+    Returns:
+        instance_data (dict): returns a dict container instance type, indicator, location, and barcode for the instance.
+    """
+    instance_data = {}
+    instance_data["instance_type"] = type
+    instance_data["indicator"] = get_container_indicators(instance)
+    if type == "digital_object":
+        digital_object = instance.get("digital_object").get("_resolved")
+        instance_data["location"] = get_file_versions(digital_object)
+        instance_data["barcode"] = None
+    else:
+        top_container = instance.get("sub_container").get("top_container").get("_resolved")
+        instance_data["location"] = get_location(top_container)
+        instance_data["barcode"] = top_container.get("barcode")
+    return instance_data
+
+
+def get_preferred_format(instances):
+    """Iterates over instances and gets the preferred delivery format based on delivery formats set in the settings.
+
+    Args:
+        instances (list): a list of ArchivesSpace instance dictionaries.
+
+    Returns:
+        preferred (list): a list of string instance information retrieved by get_instance_data.
+    """
+    instance_types = []
+    [instance_types.append((i["instance_type"])) for i in instances]
```

Comment:
Again, I think you meant:

```suggestion
    instance_types = [i["instance_type"] for i in instances]
```

Reply:
Although...I'm wondering why this. `instance_types` variable is even being set because it's not used...

Reply:
I'm actually using it in the if statements immediately below. `if "digital_object" in instances` wasn't working to return anything.